Sources/TuistGraph/Models/ProjectOptions.swift

```diff
@@ -0,0 +1,117 @@
+import Foundation
+
+extension Project {
+    /// Additional options related to the `Project`
+    public struct Options: Codable, Hashable {
+        /// Defines how to generate automatic schemes
+        public let automaticSchemesOptions: AutomaticSchemesOptions
+
+        /// Disables generating Bundle accessors.
+        public let disableBundleAccessors: Bool
+
+        /// Disable the synthesized resource accessors generation
+        public let disableSynthesizedResourceAccessors: Bool
+
+        /// Text settings to override user ones for current project
+        public let textSettings: TextSettings
+
+        public init(
+            automaticSchemesOptions: AutomaticSchemesOptions,
+            disableBundleAccessors: Bool,
+            disableSynthesizedResourceAccessors: Bool,
+            textSettings: TextSettings
+        ) {
+            self.automaticSchemesOptions = automaticSchemesOptions
+            self.disableBundleAccessors = disableBundleAccessors
+            self.disableSynthesizedResourceAccessors = disableSynthesizedResourceAccessors
+            self.textSettings = textSettings
+        }
+    }
+}
+
+// MARK: - AutomaticSchemesOptions
+
+extension Project.Options {
+    /// The automatic schemes options
+    public enum AutomaticSchemesOptions: Codable, Hashable {
+        /// Defines how to group targets into scheme for autogenerated schemes
+        public enum TargetSchemesGrouping: Codable, Hashable {
+            /// Generate a single scheme per project
+            case singleScheme
+
+            /// Group the targets according to their name suffixes
+            case byNameSuffix(build: Set<String>, test: Set<String>, run: Set<String>)
+
+            /// Do not group targets, create a scheme for each target
+            case notGrouped
+        }
+
+        /// Enable autogenerated schemes
+        case enabled(
+            targetSchemesGrouping: TargetSchemesGrouping,
+            codeCoverageEnabled: Bool,
+            testingOptions: TestingOptions
+        )
+
+        /// Disable autogenerated schemes
+        case disabled
+    }
+
+    /// The text settings options
+    public struct TextSettings: Codable, Hashable {
+        /// Whether tabs should be used instead of spaces
+        public let usesTabs: Bool?
+
+        /// The width of space indent
+        public let indentWidth: UInt?
+
+        /// The width of tab indent
+        public let tabWidth: UInt?
+
+        /// Whether lines should be wrapped or not
+        public let wrapsLines: Bool?
+
+        public init(
+            usesTabs: Bool?,
+            indentWidth: UInt?,
+            tabWidth: UInt?,
+            wrapsLines: Bool?
+        ) {
```

Comment:
This could use default values like the ProjectDescription version has, so you can shorten the `textSettings: .init(usesTabs: nil, indentWidth: nil, tabWidth: nil, wrapsLines: nil)` seen in multiple places.

Reply:
I don't like default in TuistGraph. Instead, we can define static `test` method to avoid that